presto-parser/src/main/java/com/facebook/presto/sql/tree/ExpressionTreeRewriter.java

```diff
@@ -598,11 +601,13 @@ protected Expression visitBindExpression(BindExpression node, Context<C> context
                 }
             }
 
-            Expression value = rewrite(node.getValue(), context.get());
+            List<Expression> values = node.getValues().stream()
+                    .map(value -> rewrite(value, context.get()))
+                    .collect(toImmutableList());
             Expression function = rewrite(node.getFunction(), context.get());
 
-            if ((value != node.getValue()) || (function != node.getFunction())) {
-                return new BindExpression(value, function);
+            if ((!Objects.equals(values, node.getValues())) || (function != node.getFunction())) {
```

Comment:
This changes the semantics of comparison. Use `sameElements` (instead of `Objects.equals`) in this class.